### DISCUSSION TITLE
MS SQL Server Support

### DISCUSSION THREAD

**Discussion by StephenStrickland** [keystonejs/keystone#7980] [Feature requests]
Hey Team,

Is there any interest and or plans for Keystone to support Microsoft SQL Server?

**Comment by flexdinesh** [comment]
It's currently not in our roadmap @StephenStrickland. We're hoping sometime in the future we'll be able to support it but not at the moment.

**Reply by StephenStrickland** [reply]
Would you be interested in a PR?

**Reply by flexdinesh** [reply]
Thanks for offering to contribute @StephenStrickland. As much as we'd like to support Microsoft SQL Server, adding a new database support comes with its maintenance cost. We maintain dedicated test suites for each database provider to make sure the subtle differences in behaviours between database providers stay consistent while using Keystone APIs. This means adding a new database provider adds to the maintenance cost for every feature or bug fix. We're not looking to support one more database at the moment. When we do we will revisit this discussion. Thanks again!

**Reply by dcousens** [reply]
Hi @StephenStrickland,
We are absolutely open to pull requests, but as mentioned by @flexdinesh the problem is the database specific behaviours that need to be aligned with the functionality that we have already.
If the behaviour doesn't align, we then need to document that behaviour as something for our users to watch out for.
This can make adding support a little slower :snail: than only submitting a working implementation; as the subtleties might not be easy to find.

For example
> By default, Postgres is case sensitive, while MySQL is not case sensitive, and SQLite is not case sensitive for `contains`, `startsWith` and `endsWith`.

https://keystonejs.com/docs/guides/choosing-a-database#case-sensitivity

If you're open to that, by all means we'd love your contribution! :blue_heart:
